packages/react/src/components/InAppMessaging/hooks/useMessageProps/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  MessageComponentBaseProps,
+  MessageTextAlign,
+} from '@aws-amplify/ui-react-core';
+import { StyleParams } from '../types';
+
+import { getPayloadStyle, getMessageStyles } from '../utils';
+
+describe('getPayloadStyle', () => {
+  it('returns the expected output in the happy path', () => {
+    const output = getPayloadStyle({
+      body: { style: { textAlign: 'left' as MessageTextAlign } },
+      container: { style: { backgroundColor: 'lightgray', borderRadius: 2 } },
+      header: { style: { textAlign: 'center' as MessageTextAlign } },
+      primaryButton: { style: { backgroundColor: 'salmon', color: 'olive' } },
+      secondaryButton: { style: { backgroundColor: 'sand', color: 'peru' } },
+    } as MessageComponentBaseProps);
+
+    expect(output).toMatchSnapshot();
+  });
+});
+
+describe('getMessageStyles', () => {
+  const payloadStyle: StyleParams['payloadStyle'] = {
+    body: { textAlign: 'left' as MessageTextAlign },
+    container: { backgroundColor: 'lightgray', borderRadius: 2 },
+    header: { textAlign: 'center' as MessageTextAlign },
+    primaryButton: { backgroundColor: 'salmon', color: 'olive' },
+    secondaryButton: { backgroundColor: 'sand', color: 'peru' },
+  };
+
+  const overrideStyle = {
+    body: { color: 'white' },
+    closeIconButton: { backgroundColor: 'turquoise', color: 'darkcyan' },
+    container: { backgroundColor: 'lawngreen', borderRadius: 3 },
+    header: { backgroundColor: 'lightpink' },
+    image: { backgroundColor: 'royalblue' },
+    primaryButton: { backgroundColor: 'seagreen', color: 'black' },
+    secondaryButton: { backgroundColor: 'sienna', color: 'orchid' },
+  };
+
+  it('returns the expected output in the happy path', () => {
+    const output = getMessageStyles({
+      styleParams: { payloadStyle, overrideStyle },
+    });
+
+    expect(output).toMatchSnapshot();
```

Comment:
nit: rather than rely on comparing the snapshot to the test code and relying on humans not to just update the snapshot, what do you think of expecting it to match an `expectedStyle` object we can read right here. That would make me more confident in the test.

Reply:
I think part of why we just opted for snapshots here was it was already the case with the RN versions of these utils. Imho, we *should* lean on snapshot tests for a quick (and often large) tree output and trust that our contributors respect them and make updates only after they have validated the changes are intentional. Personally, I find that to be the exact same exercise as just updating some `expectedStyle` object but I *do* understand your concern. But I will add a task to re-evaluate this approach holistically and swap out snapshot tests all at once if the team is unconvinced 👍

Reply:
That's fair.